### ISSUE TITLE
Local tunnel not working

### ISSUE THREAD

**Issue by Flaskmor** [TheLastBen/fast-stable-diffusion#1472]
After i select local tunnel nothing ho wrong like the usual but when i try to open the link (that look like the usual) it keep saying connection denied, my connection work could someone help me?

**Comment by TheLastBen** [comment]
Use gradio server when localtunnel is down

**Comment by Flaskmor** [comment]
Yeah but local tunnel is the only thing that i can use to connect to things like Hua painting? Because it's 5/6 day that it's not working for me at least

**Comment by Daviljoe193** [comment]
Ngrok support would be nice for situations like this. With the localtunnel option turned off, and if you have an Ngrok token, you can (For now) add the following cell to your notebook, which will run, spit out your tunneled URL (Which despite Painthua not saying it, supports Ngrok), and will work once you also get your usual Gradio link. This only needs to be run once.

`!pip install pyngrok`
`!ngrok authtoken NGROK_TOKEN`
`!nohup ngrok http 7860 --log=stdout &`
`!sleep 2`
`!tail -n 1 nohup.out | sed 's/.*url=//'`

**Comment by ansel44** [comment]
> Yeah but local tunnel is the only thing that i can use to connect to things like Hua painting? Because it's 5/6 day that it's not working for me at least

have you solved the problem? how to use local tunnel again?
